index.html

```diff
@@ -174,95 +173,117 @@
       </div>
       <section>
         <h2>
-          Syntax
+          Validation
         </h2>
         <p>
-          When URLs are used for payment method identifiers they MUST be
-          absolute URLs including, at most, a <a>scheme</a>, <a>host</a>,
-          <a>port</a> and <a>path</a>. The URL must be a <a>potentially
-          trustworthy URL</a> as defined in the [[SECURE-CONTEXTS]]
-          specification. Identifier URLs MUST have null <a>query</a> and
-          <a>fragment</a> values.
+          The steps to <dfn data-lt="validate|valid">validate a URL-based
+          payment method identifier</dfn> are given by the following algorithm.
+          The algorithm takes a <var>potential URL</var> <a>DOMString</a> as
+          input and return either true or an error:
         </p>
+        <ol class="algorithm">
+          <li>Let <var>url</var> be the result of trying to <a data-lt=
+          "URL Parser">parse</a> <var>potential URL</var>. If failure,
+          <a>throw</a> a <code>TypeError</code> exception.
+          </li>
+          <li>If <a>scheme</a> is not "https", <a>throw</a> a
+          <code>TypeError</code> exception.
+          </li>
+          <li>If <a>query</a> is not null or <a>fragment</a> is not null,
+          <a>throw</a> <code>TypeError</code> exception.
+          </li>
+          <li>Return true.
+          </li>
+        </ol>
       </section>
       <section>
         <h2>
           Matching
         </h2>
         <p>
-          User agents MUST use the following algorithm to determine whether two
-          payment method identifiers that are URLs match:
+          The steps to <dfn>determine whether two payment method identifiers
+          match</dfn> is given by the following algorithm. The algorithm takes
+          two string <var>A</var> and <var>B</var>.
         </p>
-        <ul>
-          <li>Let <em>A</em> be the first payment method identifier string and
-          let <em>B</em> be the second payment method identifier string.
-          </li>
-          <li>Let <em>urlA</em> be the result from the <a>URL parser</a> when
-          parsing <em>A</em> with no <a>base URL</a>.
+        <ol class="algorithm">
+          <li>Try to <a>parse</a> <var>A</var> and let <var>urlA</var> be the
```

Comment:
Okay. You used "URL parser" as well earlier. Seems weird to use different referencing terms within the same document.